### ISSUE TITLE
protocols in routes should be configurable

### ISSUE THREAD

**Issue by ArekCzarnik** [Kong/kubernetes-ingress-controller#52]
**Is this a BUG REPORT or FEATURE REQUEST?**:

**Kong Ingress controller version**:
0.0.3

**Kubernetes version** (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.2", GitCommit:"5fa2db2bd46ac79e5e00a4e6ed24191080aa463b", GitTreeState:"clean", BuildDate:"2018-01-18T21:12:46Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.2", GitCommit:"81753b10df112992bf51bbc2c2f85208aad78335", GitTreeState:"clean", BuildDate:"2018-04-27T09:10:24Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}

**Environment**:

- **Cloud provider or hardware configuration**: 
- **OS** (e.g. from /etc/os-release): Linux 
- **Kernel** (e.g. `uname -a`):

**What happened**:
we create a  KongIngress yaml resource with routes, that with have a protocols section (https), but the ingress controller disregard this value.

**What you expected to happen**:
protocols in routes should be configurable.

**How to reproduce it** (as minimally and precisely as possible):
deploy and setup kongingress resource:

```apiVersion: configuration.konghq.com/v1
kind: KongIngress
metadata:
  name:deployment-example
route:
  protocols:
  - https
  methods:
  - GET
  preserve_host: false
  regex_priority: 0
  strip_path: true
```

**Comment by dag24** [comment]
+1, I think specifying this should follow the `https_only` behaviour of the `/apis` endpoint.

**Comment by gerred** [comment]
@ArekCzarnik Reviewing this, will be getting this in this week.

**Comment by gerred** [comment]
Closed as duplicate of #64
